template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java

```diff
@@ -23,64 +21,47 @@
-                break;
-            default:
-                break;
-        }
-
-        return roleConfigs;
-    }
-
-    @Override
-    protected List<ApiClusterTemplateVariable> getVariables(String roleType, HostgroupView hostGroupView, TemplatePreparationObject templatePreparationObject) {
-        List<ApiClusterTemplateVariable> variables = new ArrayList<>();
-
-        switch (roleType) {
-            case "RANGER_ADMIN":
-                RDSConfig rdsConfig = getRDSConfig(templatePreparationObject);
+            case RangerRoles.RANGER_ADMIN:
+                RDSConfig rdsConfig = getRDSConfig(source);
                 RdsView rangerRdsView = new RdsView(rdsConfig);
-                if (rangerRdsView.getDatabaseVendor() == DatabaseVendor.POSTGRES) {
-                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_host").value(rangerRdsView.getHost()));
-                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_name").value(rangerRdsView.getDatabaseName()));
-                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_type").value("PostgreSQL"));
-                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_user").value(rangerRdsView.getConnectionUserName()));
-                    variables.add(new ApiClusterTemplateVariable().name("ranger-ranger_database_password").value(rangerRdsView.getConnectionPassword()));
-                } else {
-                    throw new CloudbreakServiceException("Ranger database supports only PostgreSQL");
-                }
-                break;
+                return List.of(
+                        config("ranger_database_host", rangerRdsView.getHost()),
+                        config("ranger_database_name", rangerRdsView.getDatabaseName()),
+                        config("ranger_database_type", getRangerDbType(rangerRdsView)),
+                        config("ranger_database_user", rangerRdsView.getConnectionUserName()),
+                        config("ranger_database_password", rangerRdsView.getConnectionPassword())
+                );
             default:
-                break;
+                return List.of();
         }
-
-        return variables;
     }
 
     @Override
     public String getServiceType() {
-        return "RANGER";
+        return RangerRoles.RANGER;
     }
 
     @Override
     public List<String> getRoleTypes() {
-        return Collections.singletonList("RANGER_ADMIN");
+        return List.of(RangerRoles.RANGER_ADMIN);
     }
 
     @Override
     public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
         return cmTemplateProcessor.isRoleTypePresentInService(getServiceType(), getRoleTypes());
     }
 
+    private static String getRangerDbType(RdsView rdsView) {
```

Comment:
I guess this should not be static

Reply:
I prefer `static` for methods that don't need instance data (ie. kind of "local" utilities).

Reply:
This is not very common in cloudbreak code. We usually don't use statics except in public utility classes.